### ISSUE TITLE
FireTriggers Option

### ISSUE THREAD

**Issue by thomasonb** [borisdj/EFCore.BulkExtensions#38]
I needed the ability to fire triggers on my tables when doing bulk inserts. I don't actually know how to do a pull request, but below is the code (Seperated by file) that I modified to allow me to do that if anyone else needs to ability too. It would be nice if this were in the package. This also gives the ability for All the SQLBulkCopyOptions. Essentially I added a BulkCopyOptions into BulkConfig and anywhere BulkConfig was being used I modified it to accept BulkCopyOptions for a parameter. _Note, I did not remove the KeepIdentity property which is a SQLBulkCopyOptions option, but there are now 0 references too it. I also left the original code (commented out) in GetSqlBulkCopy() simply because I only tested it for my scenario and wasn't sure if it followed the same logic path that is expected for other scenarios._

**BulkConfig.cs**
```cs
using System.Collections.Generic;
using System.Data.SqlClient;

namespace EFCore.BulkExtensions
{
    public class BulkConfig
    {
        public bool PreserveInsertOrder { get; set; }

        public bool SetOutputIdentity { get; set; }

        public int BatchSize { get; set; } = 2000;

        public int? NotifyAfter { get; set; }

        public int? BulkCopyTimeout { get; set; }

        public bool EnableStreaming { get; set; }

        public bool UseTempDB { get; set; }

        public bool KeepIdentity { get; set; }

        public List<string> PropertiesToInclude { get; set; }

        public List<string> PropertiesToExclude { get; set; }

        public List<string> UpdateByProperties { get; set; }

        public SqlBulkCopyOptions BulkCopyOptions { get; set; }
    }
}
```

**SqlBulkOperation.cs**
```cs
public static void Insert<T>(DbContext context, IList<T> entities, TableInfo tableInfo, Action<decimal> progress)
        {
            var sqlConnection = OpenAndGetSqlConnection(context);
            var transaction = context.Database.CurrentTransaction;
            try
            {
                using (var sqlBulkCopy = GetSqlBulkCopy(sqlConnection, transaction, tableInfo.BulkConfig.BulkCopyOptions))
                {
                    tableInfo.SetSqlBulkCopyConfig(sqlBulkCopy, entities, progress);
                    using (var reader = ObjectReaderEx.Create(entities, tableInfo.ShadowProperties, context, tableInfo.PropertyColumnNamesDict.Keys.ToArray()))
                    {
                        sqlBulkCopy.WriteToServer(reader);
                    }
                }
            }
            finally
            {
                if (transaction == null)
                {
                    sqlConnection.Close();
                }
            }
        }

        public static async Task InsertAsync<T>(DbContext context, IList<T> entities, TableInfo tableInfo, Action<decimal> progress)
        {
            var sqlConnection = await OpenAndGetSqlConnectionAsync(context);
            var transaction = context.Database.CurrentTransaction;
            try
            {
                using (var sqlBulkCopy = GetSqlBulkCopy(sqlConnection, transaction, tableInfo.BulkConfig.BulkCopyOptions))
                {
                    tableInfo.SetSqlBulkCopyConfig(sqlBulkCopy, entities, progress);
                    using (var reader = ObjectReaderEx.Create(entities, tableInfo.ShadowProperties, context, tableInfo.PropertyColumnNamesDict.Keys.ToArray()))
                    {
                        await sqlBulkCopy.WriteToServerAsync(reader).ConfigureAwait(false);
                    }
                }
            }
            finally
            {
                if (transaction == null)
                {
                    sqlConnection.Close();
                }
            }
        }

private static SqlBulkCopy GetSqlBulkCopy(SqlConnection sqlConnection, IDbContextTransaction transaction, SqlBulkCopyOptions options = SqlBulkCopyOptions.Default)
        {
            if (transaction == null)
            {
                return new SqlBulkCopy(sqlConnection, options, null);
            }
            else
            {
                var sqlTransaction = (SqlTransaction)transaction.GetDbTransaction();
                return new SqlBulkCopy(sqlConnection, options, sqlTransaction);
            }
            //if (transaction == null)
            //{
            //    if (keepIdentity)
            //        return new SqlBulkCopy(sqlConnection, SqlBulkCopyOptions.KeepIdentity, null);
            //    else
            //        return new SqlBulkCopy(sqlConnection);
            //}
            //else
            //{
            //    var sqlTransaction = (SqlTransaction)transaction.GetDbTransaction();
            //    if (keepIdentity)
            //        return new SqlBulkCopy(sqlConnection, SqlBulkCopyOptions.KeepIdentity, sqlTransaction);
            //    else
            //        return new SqlBulkCopy(sqlConnection, SqlBulkCopyOptions.Default, sqlTransaction);
            //}
        }
```

**Comment by borisdj** [comment]
BulkCopyOptions added to BulkConfig and KeepIdentity removed.
Updated info in README.
Published NuGet 2.0.6

**Comment by borisdj** [comment]
@thomasonb regarding pull requests, in [CONTRIBUTING.md](https://github.com/borisdj/EFCore.BulkExtensions/blob/master/CONTRIBUTING.md) file there is link to [pull requests](https://help.github.com/articles/about-pull-requests/) info.
